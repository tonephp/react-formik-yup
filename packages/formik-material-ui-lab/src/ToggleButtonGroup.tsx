import * as React from 'react';
import MuiToggleButtonGroup, {
  ToggleButtonGroupProps as MuiToggleButtonGroupProps,
} from '@material-ui/lab/ToggleButtonGroup';
// import ToggleButton from '@material-ui/lab/ToggleButton';
import { FieldProps } from 'formik';
import invariant from 'tiny-warning';

export interface ToggleButtonGroupProps
  extends FieldProps,
    Omit<MuiToggleButtonGroupProps, 'name' | 'value' | 'error'> {
  type?: string;
}

export function fieldToToggleButtonGroup({
  field: { onChange: _onChange, onBlur: fieldOnBlur, ...field },
  type,
  onChange,
  onBlur,
  form,
  ...props
}: ToggleButtonGroupProps): MuiToggleButtonGroupProps {
  if (process.env.NODE_ENV !== 'production') {
    invariant(
      type === 'checkbox',
      `property type=checkbox is missing from field ${field.name}, this can caused unexpected behaviour`
    );

    if (!props.exclusive) {
      invariant(
        Array.isArray(field.value),
        `value for ${field.name} is not an array, this can caused unexpected behaviour`
      );
    }
  }

  return {
    onBlur:
      onBlur ??
      function () {
        fieldOnBlur(field.name);
      },
    onChange:
      onChange ??
      function (_event, newValue) {
        form.setFieldValue(field.name, newValue);
      },
    ...field,
    ...props,
  };
}

export function ToggleButtonGroup(props: ToggleButtonGroupProps) {
  return <MuiToggleButtonGroup {...fieldToToggleButtonGroup(props)} />;
}

ToggleButtonGroup.displayName = 'FormikMaterialUIToggleButtonGroup';
