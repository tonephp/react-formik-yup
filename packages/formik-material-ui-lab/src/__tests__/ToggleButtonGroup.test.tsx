import React from 'react';
import { test, expect } from '@jest/globals';
import { Field } from 'formik';

import { render } from './utils';

import { ToggleButtonGroup } from '../ToggleButtonGroup';

test('renders', async () => {
  // @ts-ignore
  const onSubmit = jest.fn();
  const { asFragment } = render(
    <Field component={ToggleButtonGroup} name="toggle" type="checkbox" />,
    {
      initialValues: {
        toggle: [],
      },
      onSubmit,
    }
  );

  expect(asFragment()).toMatchSnapshot();
});
