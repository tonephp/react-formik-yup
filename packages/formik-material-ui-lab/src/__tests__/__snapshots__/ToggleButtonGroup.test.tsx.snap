// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <form
    action="#"
    data-testid="form"
  >
    <div
      class="MuiToggleButtonGroup-root"
      name="toggle"
      role="group"
    />
    <button
      data-testid="submit"
    >
      submit
    </button>
  </form>
</DocumentFragment>
`;
